src/openfl/display/DisplayObject.hx

```diff
@@ -1256,7 +1258,14 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable #if openf
 						
 					}
 					
-					__cacheBitmap.bitmapData = bitmapData;
+					if (__cacheBitmapData != bitmapData) {
+						
+						__cacheBitmapData = bitmapData;
+						@:privateAccess __cacheBitmapData.__scale = scale;
```

Comment:
set `__scale` right after creation of `BitmapData`s

Reply:
Yes please!